src/lambda/models/samLambdaRuntime.ts

```diff
@@ -106,6 +128,25 @@ export function getFamily(runtime: string): RuntimeFamily {
     return RuntimeFamily.Unknown
 }
 
+function handleDeprecatedRuntime(runtime: Runtime) {
+    const moreInfo = localize('AWS.generic.message.learnMore', 'Learn More')
+    vscode.window
+        .showErrorMessage(
+            localize(
+                'AWS.samcli.deprecatedRuntime',
+                'Runtime {0} has been deprecated. Please update to a currently-supported runtime.',
```

Comment:
We should be clear about who is doing the deprecation. Right now is seems like the toolkit has deprecated the runtime, which would be weird since throwing an error isn't deprecation, it's just straight up dropping support. At the very least update the error message to be clear that the toolkit no longer supports the runtime _because_ it has been deprecated by Lambda.

Reply:
I feel like the "More Info" button accomplishes this. This is short and to the point and informs the user that it's not going to work. This is better than what we had before, where we would just say it's an "unsupported runtime".

Reply:
> This is short and to the point and informs the user that it's not going to work.

Agreed, though I think including the actor in the first sentence is important, otherwise users need to click a link to figure it out:

```
Runtime {0} has been deprecated by Lambda.
```

This also sets user expectations about the provided link.